internal/dag/conditions.go

```diff
@@ -30,13 +31,21 @@ func pathCondition(conds []projcontour.Condition) Condition {
 	}
 }
 
-func pathConditionsValid(conds []projcontour.Condition) bool {
+// pathConditionsValid validates a slice of Conditions.
+// It encodes the business rules about what is allowed for prefix Conditions.
+func pathConditionsValid(conds []projcontour.Condition, sw *ObjectStatusWriter, conditionsContext string) bool {
 	found := 0
 	for _, cond := range conds {
 		if cond.Prefix != "" {
+			// We found a prefix, so increment the prefix counter.
```

Comment:
I don't think this commend adds a lot. Perhaps something like renaming `found` to `prefixCount` would make the code more self explanatory.

Reply:
I actually started doing that when I was rewriting into a single verify function, and forgot to bring it back when I reverted. Agreed, I'll do that now.